src/algokit/cli/tasks/sign_transaction.py

```diff
@@ -0,0 +1,63 @@
+import json
+from pathlib import Path
+from typing import cast
+
+import click
+from algosdk import encoding
+from algosdk.transaction import Transaction, retrieve_from_file, write_to_file
+
+from algokit.cli.tasks.utils import get_account_with_private_key
+
+
+def get_transaction(file: Path | None, transaction: str | None) -> Transaction:
+    if file:
+        txns: list[Transaction] = retrieve_from_file(str(file))  # type: ignore[no-untyped-call]
+        if len(txns) > 1:
+            raise click.ClickException("Only one transaction per file is supported.")
+        return txns[0]
+    elif transaction:
+        return cast(Transaction, encoding.msgpack_decode(transaction))  # type: ignore[no-untyped-call]
+    else:
+        raise click.ClickException("Provide either a file to sign or a transaction to sign, not both or none.")
+
+
+def confirm_transaction(txn: Transaction) -> bool:
+    click.echo(json.dumps(txn.__dict__, indent=2))
+    response = click.prompt("Do you want to sign the above transaction?", type=click.Choice(["y", "n"]), default="n")
+    return bool(response == "y")
+
+
+def sign_and_output_transaction(txn: Transaction, private_key: str, output: Path | None) -> None:
+    signed_txn = txn.sign(private_key)  # type: ignore[no-untyped-call]
+
+    if output:
+        write_to_file([signed_txn], str(output))  # type: ignore[no-untyped-call]
+        click.echo(f"Signed transaction written to {output}")
+    else:
+        click.echo(encoding.msgpack_encode({"txn": signed_txn.dictify()}))  # type: ignore[no-untyped-call]
+
+
+@click.command(name="sign", help="Sign an Algorand transaction.")
```

Comment:
@robdmoore just fyi, id suggest to use this as a starting point for pr review, remaining tweaks in other tasks are just minor polishing